diff --git a/index.d.ts b/index.d.ts
index 7f1a205e4a09eed1eff01577e345ee5ada6cb084..5ecc0c49801b5acf142f5713a792b080bf19a1c0 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -47,7 +47,7 @@ export class addonBuilder {
      * Docs: https://github.com/Stremio/stremio-addon-sdk/blob/master/docs/api/requests/defineStreamHandler.md
      */
     defineStreamHandler(
-        handler: (args: { type: ContentType; id: string }) => Promise<{ streams: Stream[] } & Cache>,
+        handler: (args: { type: ContentType; id: string, req: any, config?: any }) => Promise<{ streams: Stream[] } & Cache>,
     ): void;
 
     /**
@@ -516,7 +516,10 @@ export interface Manifest {
          * Used to provide an adequate warning to the user.
          */
         p2p?: boolean | undefined;
+        configurable?: boolean | undefined;
+        configurationRequired?: boolean | undefined;
     } | undefined;
+    config?: any | undefined;
 }
 
 /**
@@ -645,4 +648,4 @@ export function serveHTTP(
          */
         static?: string | undefined;
     },
-): void;
+): Promise<{url: string, server: any, app: any}>;
